atlante/template/grating/grating.go

```diff
@@ -0,0 +1,177 @@
+	}
+	if NumberOfCols < MinRowCol || NumberOfCols > MaxRowCol {
+		return nil, fmt.Errorf("Invalid number of cols: %v; cols must be between %v and %v", NumberOfCols, MinRowCol, MaxRowCol)
+	}
+
+	colOffset := width / float64(NumberOfCols)
+	rowOffset := height / float64(NumberOfRows)
+	extent := geom.Extent{
+		x, y, x + width, y + height,
+	}
+
+	if flipY {
+		extent[1], extent[3] = extent[3], extent[1]
+	}
+	return &Grating{
+		Rows: NumberOfRows,
+		Cols: NumberOfCols,
+
+		Extent: extent,
+
+		FlipY:  flipY,
+		Width:  colOffset,
+		Height: rowOffset,
+	}, nil
+
+}
+
+func (grate *Grating) labelForRow(row int) string {
+	var (
+		buff   strings.Builder
+		lblLen = len(LabelLetters)
+	)
+	if row < lblLen {
+		return LabelLetters[row]
+	}
+	prelbl := grate.labelForRow(row / lblLen)
+	buff.WriteString(prelbl)
+	buff.WriteString(LabelLetters[row%lblLen])
+	return buff.String()
+}
+func (grate *Grating) LabelForRow(row int) string {
+	if row < 0 || row > int(grate.Rows) {
+		return ""
+	}
+	var (
+		buff   strings.Builder
+		lblLen = len(LabelLetters)
+	)
+	if row < lblLen {
+		if int(grate.Rows) >= lblLen {
+			// Need to add the first letter
+			for i := 1; i < int(grate.Rows)/lblLen; i++ {
+				buff.WriteString(LabelLetters[0])
+			}
+		}
+		buff.WriteString(LabelLetters[row])
+		return buff.String()
+	}
+	return grate.labelForRow(row)
+}
+
+func (grate *Grating) LabelForCol(col int) string {
+	if col < 0 || col > int(grate.Cols) {
+		return ""
+	}
+	return fmt.Sprintf("%d", col+1)
+}
+
+// LineForRow returns the row from 0 — Rows-1
+func (grate *Grating) LineForRow(row uint) geom.Line {
+	rowOffset := (float64(row) * grate.Height)
+	if grate.FlipY {
+		rowOffset *= -1
+	}
+	return geom.Line{
+		[2]float64{
+			grate.Extent[0],
+			grate.Extent[1] + rowOffset,
+		},
+		[2]float64{
+			grate.Extent[2],
+			grate.Extent[1] + rowOffset,
+		},
+	}
+
+}
+
+// LineForCol returns the column from 0 — Cols-1
+func (grate *Grating) LineForCol(col uint) geom.Line {
+	colOffset := (float64(col) * grate.Width)
+	return geom.Line{
+		[2]float64{
+			grate.Extent[0] + colOffset,
+			grate.Extent[1],
+		},
+		[2]float64{
+			grate.Extent[0] + colOffset,
+			grate.Extent[3],
+		},
+	}
+}
+
+func (grate *Grating) YForRow(row int) float64 {
+	if grate == nil {
+		return 0.0
+	}
+	rowOffset := (float64(row) * grate.Height)
+	if grate.FlipY {
+		rowOffset *= -1
+	}
+	return grate.Extent[1] + rowOffset
+}
+func (grate *Grating) XForCol(col int) float64 {
+	if grate == nil {
+		return 0.0
+	}
+	colOffset := (float64(col) * grate.Width)
+	return grate.Extent[0] + colOffset
+}
+
+// PositionFor row and col,
+func (grate *Grating) PositionFor(row, col int) geom.Point {
+	return geom.Point{
+		grate.XForCol(col),
+		grate.YForRow(row),
+	}
+	rowOffset := (float64(row) * grate.Height)
+	if grate.FlipY {
+		rowOffset *= -1
+	}
+	colOffset := (float64(col) * grate.Width)
+	return geom.Point{
+		grate.Extent[0] + colOffset,
+		grate.Extent[1] + rowOffset,
+	}
```

Comment:
Looks like the initial return is causing the rest of this code to never be executed.

Reply:
Yup, forgot to remove the old code.